src/bilinear_linearization.jl

```diff
@@ -0,0 +1,811 @@
+#=
```

Comment:
we really need small test cases for all corner cases.

Reply:
I am adding more tests and trying to meet the codecov requirements